zkevm-circuits/src/state_circuit.rs

```diff
@@ -69,8 +213,8 @@ pub struct SortKeysConfig {
 type Lookup<F> = (&'static str, Expression<F>, Expression<F>);
 
 /// State Circuit for proving RwTable is valid
-#[derive(Default)]
-pub struct StateCircuit<F: Field> {
+#[derive(Default, Clone)]
+pub struct StateCircuit<F> {
     pub(crate) randomness: F,
     pub(crate) rows: Vec<Rw>,
     pub(crate) n_rows: usize,
```

Comment:
one question:  Does `n_rows` need to be equal to evm circui rows number in real proof ?

Reply:
no. n_rows is the rows of rw table. rw table in evm circuit and state circuit should be same.   Not related to the total rows (max of fixed_table, rw_table, tx_table, other parts..) of circuit.